base/linalg/lapack.jl

```diff
@@ -31,6 +31,12 @@ macro chkuplo()
       throw(ArgumentError(string("uplo argument must be 'U' (upper) or 'L' (lower), got ", repr(uplo)))))
 end
 
+#Check that {c}transpose is correctly specified
+macro chktrans()
+    :((trans=='N' || trans=='C' || trans=='T') ||
+      throw(ArgumentError(string("trans argument must be 'N' (no transpose), 'T' (transpose), or 'C' (conjugate transpose), got ", repr(trans)))))
+end
```

Comment:
I'd be happy to change both `chktrans` and `chkuplo` to functions as part of this PR :smile: